src/EnergyPlus/HeatingCoils.cc

```diff
@@ -3320,6 +3340,74 @@ namespace HeatingCoils {
 
 	}
 
+	void
+	SetHeatingCoilAsDesicRegenCoil(
+		std::string const & CoilType, // must match coil types in this module
+		std::string const & CoilName, // must match coil names for the coil type
+		int & DesiccantDehumIndex, // index of desiccant dehumidifier
+		bool & ErrorsFound // set to true if problem
+		) {
+
+		// FUNCTION INFORMATION:
+		//       AUTHOR         Bereket Nigusse
+		//       DATE WRITTEN   February 2016
+		//       MODIFIED       na
+		//       RE-ENGINEERED  na
+
+		// PURPOSE OF THIS FUNCTION:
+		// This function looks up the given coil and registers the coil as desiccant regeneration air heating coil.
+		// If incorrect coil type or name is given, ErrorsFound is returned as true
+
+		// METHODOLOGY EMPLOYED:
+		// na
+
+		// REFERENCES:
+		// na
+
+		// Using/Aliasing
+		using InputProcessor::FindItem;
+
+		// Return value
+		// na
+
+		// Locals
+		// FUNCTION ARGUMENT DEFINITIONS:
+
+		// FUNCTION PARAMETER DEFINITIONS:
+		// na
+
+		// INTERFACE BLOCK SPECIFICATIONS:
+		// na
+
+		// DERIVED TYPE DEFINITIONS:
+		// na
+
+		// FUNCTION LOCAL VARIABLE DECLARATIONS:
+		int WhichCoil;
+		int FoundType; // Integer equivalent of coil type
+
+		// Obtains and Allocates HeatingCoil related parameters from input file
+		if ( GetCoilsInputFlag ) { //First time subroutine has been entered
+			GetHeatingCoilInput();
+			GetCoilsInputFlag = false;
+		}
+
+		WhichCoil = 0;
+		FoundType = FindItem( CoilType, cAllCoilTypes, NumAllCoilTypes );
```

Comment:
The Type/Name functions are required when the index is unknown. If the index is known, the function could be written to handle all other logic (e.g., if type is electric or gas or desuperheater). Not sure mining functions will pose a significant runtime penalty by passing strings, but every clock cycle saved is a clock cycle earned. I guess this all means there should only be 1 function in a module that uses Type/Name, the GetIndex function. All other mining functions should pass the index (hindsight).